cmd/deploy.go

```diff
@@ -38,16 +41,24 @@ type deployCommandConfig struct {
 }
 
 type AppsodyApplication struct {
-	APIVersion string   `yaml:"apiVersion"`
-	Kind       string   `yaml:"kind"`
-	Metadata   Metadata `yaml:"metadata"`
-	Spec       Spec     `yaml:"spec"`
+	APIVersion string                         `yaml:"apiVersion" json:"apiVersion"`
+	Kind       string                         `yaml:"kind" json:"kind"`
+	Metadata   Metadata                       `yaml:"metadata" json:"metadata"`
+	Spec       v1beta1.AppsodyApplicationSpec `yaml:"spec" json:"spec"`
```

Comment:
Explore using `v1beta1.AppsodyApplication` - (unless there's a good reason not to) - for future releases.

Reply:
Yes, I had not seen that `v1beta1.AppsodyApplicaiton` has a field for [ObjectType](https://github.com/kubernetes/apimachinery/blob/bb31b70367b776cca4a73412138411e77a36a104/pkg/apis/meta/v1/types.go#L110) which has [labels](https://github.com/kubernetes/apimachinery/blob/bb31b70367b776cca4a73412138411e77a36a104/pkg/apis/meta/v1/types.go#L231). Would need to investigate if our local `AppsodyApplication` is used somewhere else in the codebase, maybe when deploying to Knative?